src/Api/Managers/NotificationManager.cs

```diff
@@ -1,65 +1,87 @@
-                        }
+                        dbContext.Notifications.Add(new Notification() { UserId = supervisor, ChangedByUserId = changedBy, TopicId = topicId, Message = "You are now related to a new topic " + topicId + " modified by " + changedBy});
                     }
-                    else if (model.Students != null)
+                }
+                else if (model.Students != null)
+                {
+                    foreach (int student in model.Students)
                     {
-                        foreach (int student in model.Students)
-                        {
-                            dbContext.Notifications.Add(new Notification(new NotificationFormModel()) { Id = student, ChangedById = userId, TopicId = topicId, Message = "Notification received", IsReadOrNot = false });
-                        }
+                        dbContext.Notifications.Add(new Notification() { UserId = student, ChangedByUserId = changedBy, TopicId = topicId, Message = "You are now related to a new topic " + topicId + " modified by " + changedBy});
                     }
-                    else
+                }
+                else
+                {
+                    foreach (int reviewer in model.Reviewers)
                     {
-                        foreach (int reviewer in model.Reviewers)
-                        {
-                            dbContext.Notifications.Add(new Notification(new NotificationFormModel()) { Id = reviewer, ChangedById = userId, TopicId = topicId, Message = "Notification received", IsReadOrNot = false });
-                        }
+                        dbContext.Notifications.Add(new Notification() { UserId = reviewer, ChangedByUserId = changedBy, TopicId = topicId, Message = "You are now related to a new topic " + topicId + " modified by " + changedBy});
                     }
-
-                    dbContext.SaveChanges();
-                    transaction.Commit();
-                    return true;
                 }
-                catch (Exception e)
 
+                dbContext.SaveChanges();
+                return true;
+            }
+            catch (Exception e)
+
+            {
+                Console.WriteLine(e.ToString());
+                return false;
+            }
+        }
+
+        public virtual bool UpdateIsReadOrNot(int userId)
+        {
+            if (dbContext.Topics.AsNoTracking().FirstOrDefault(t => t.Id == userId) != null)
+            {
+                using (var transaction = dbContext.Database.BeginTransaction())
                 {
-                    Console.WriteLine(e.ToString());
-                    transaction.Rollback();
-                    return false;
+                    try
+                    {
+                        dbContext.Database.ExecuteSqlCommand($"update Notifications set \"IsRead\" = '{true}' where \"UserId\" = '{userId}'");
```

Comment:
Its not the userId!
And one Core-Feature of  EF that you don't have to write SQL Querys, so don't do this and use the ef.